pkg/controller/kibana/pod.go

```diff
@@ -133,27 +125,38 @@ func NewPodTemplateSpec(
 		WithAnnotations(DefaultAnnotations).
 		WithDockerImage(kb.Spec.Image, container.ImageRepository(container.KibanaImage, v)).
 		WithReadinessProbe(readinessProbe(kb.Spec.HTTP.TLS.Enabled(), basePath)).
-		WithPorts(ports).
-		WithInitContainers(initConfigContainer(kb))
+		WithPorts(ports)
 
 	for _, volume := range volumes {
 		builder.WithVolumes(volume.Volume()).WithVolumeMounts(volume.VolumeMount())
 	}
 
 	// Kibana 7.5.0 and above support running with a read-only root filesystem,
 	// but require a temporary volume to be mounted at /tmp for some reporting features
-	// and a plugin volume mounted at /usr/share/kibana/plugins.
+	// and a plugin volume mounted at /usr/share/kibana/plugins. Also needed is an
+	// init container to copy any existing plugins in /usr/share/kibana/plugins to the
+	// temporary volume.
 	// Limiting to 7.10.0 here as there was a bug in previous versions causing rebuilding
 	// of browser bundles to happen on plugin install, which would attempt a write to the
 	// root filesystem on restart.
-	if v.GTE(version.From(7, 10, 0)) && setDefaultSecurityContext {
+	var canEnableSecurityContext = v.GTE(initcontainer.HardenedSecurityContextSupportedVersion) && setDefaultSecurityContext
+	if canEnableSecurityContext {
+		scriptsConfigMapVolume := initcontainer.NewScriptsConfigMapVolume(kb.Name)
 		builder.WithContainersSecurityContext(defaultSecurityContext).
 			WithPodSecurityContext(defaultPodSecurityContext).
 			WithVolumes(LogsVolume.Volume()).WithVolumeMounts(LogsVolume.VolumeMount()).
 			WithVolumes(PluginsVolume.Volume()).WithVolumeMounts(PluginsVolume.VolumeMount()).
-			WithVolumes(TempVolume.Volume()).WithVolumeMounts(TempVolume.VolumeMount())
+			WithVolumes(TempVolume.Volume()).WithVolumeMounts(TempVolume.VolumeMount()).
+			WithVolumes(scriptsConfigMapVolume.Volume()).WithVolumeMounts(scriptsConfigMapVolume.VolumeMount())
```

Comment:
The scripts volume mount has to be present also when the security context is disabled.

Reply:
I have some unit tests to fix tomorrow, but it shouldn't block the continued review of this.